apps/mark-scan/frontend/src/pages/validate_ballot_page.tsx

```diff
@@ -1,17 +1,109 @@
 /* istanbul ignore file - placeholder component that will change */
-import { Main, Screen, Text, H1, P } from '@votingworks/ui';
+import { useContext } from 'react';
+
+import styled from 'styled-components';
+import {
+  Prose,
+  Screen,
+  H1,
+  WithScrollButtons,
+  Main,
+  Button,
+} from '@votingworks/ui';
+
+import { assert } from '@votingworks/basics';
+import { Review } from '@votingworks/mark-flow-ui';
+import {
+  getElectionDefinition,
+  getInterpretation,
+  invalidateBallot,
+  validateBallot,
+} from '../api';
+
+import { BallotContext } from '../contexts/ballot_context';
+import { ButtonFooter } from '../components/button_footer';
+import { DisplaySettingsButton } from '../components/display_settings_button';
+
+const ContentHeader = styled.div`
+  padding: 0.5rem 0.75rem 0;
+`;
+
+export function ValidateBallotPage(): JSX.Element | null {
+  const getInterpretationQuery = getInterpretation.useQuery();
+  const getElectionDefinitionQuery = getElectionDefinition.useQuery();
+  // We use the contest data stored in BallotContext but vote data from the interpreted ballot
+  const { contests, precinctId, resetBallot, endVoterSession } =
+    useContext(BallotContext);
+
+  assert(
+    typeof precinctId !== 'undefined',
+    'precinctId is required to render ValidateBallotPage'
+  );
+
+  const settingsButton = <DisplaySettingsButton />;
+
+  const invalidateBallotMutation = invalidateBallot.useMutation();
+  const validateBallotMutation = validateBallot.useMutation();
+  async function invalidateBallotCallback() {
+    invalidateBallotMutation.mutate();
+    resetBallot();
+    await endVoterSession();
```

Comment:
Hmm I thought I read something about entry actions not supporting async/await syntax but I think I misunderstood because I can't find it in the docs. [04a8f72](https://github.com/votingworks/vxsuite/pull/3855/commits/04a8f72588ee7bd01e18b6f8c0b850f694eb26ec)

Reply:
https://github.com/votingworks/vxsuite/pull/3855/commits/32a945edf677f2626105fc8624c61b92d40b5fff